kinesis.go

```diff
@@ -188,3 +196,11 @@ func (k *kinesis) getErrCount() int64 {
 func getLock(sem chan bool) {
 	sem <- true
 }
+
+func (k *kinesis) decreaseRequestLimit() {
+	k.limit = k.limit >> 1
```

Comment:
What happens if this ever reaches zero? I see we are using shift 1 which integer divides by two each time it is called and could in theory reach zero if we traverse too many small objects in a short timespan.

I know the likelihood is not high, but I do like to ask anyways :-P